lib/iris/tests/__init__.py

```diff
@@ -388,19 +388,40 @@ def assertCDL(self, netcdf_filename, reference_filename=None, flags="-h"):
         re_ncprop = re.compile(r"^\s*:_NCProperties *=")
         lines = [line for line in lines if not re_ncprop.match(line)]
 
-        # Sort the dimensions (except for the first, which can be unlimited).
-        # This gives consistent CDL across different platforms.
-        def sort_key(line):
-            return ("UNLIMITED" not in line, line)
+        if "dimensions:" in lines:
```

Comment:
Interesting fix, maybe useful to adopt generally : 
Without it, assertCDL fails on a saved  scalar cube (i.e. which has no dims)